aries_vcx/src/handlers/out_of_band/sender/sender.rs

```diff
@@ -86,3 +90,83 @@ impl OutOfBandSender {
         })
     }
 }
+
+
+#[cfg(test)]
+mod tests {
+    use crate::messages::connection::request::tests::_request;
+    use crate::handlers::connection::public_agent::tests::_public_agent;
+    use crate::messages::basic_message::message::BasicMessage;
+    use crate::messages::connection::invite::test_utils::{_pairwise_invitation, _pairwise_invitation_random_id, _public_invitation, _public_invitation_random_id};
+    use crate::messages::connection::service::FullService;
+    use crate::messages::issuance::credential_offer::CredentialOffer;
+    use crate::utils::devsetup::SetupMocks;
+
+    use super::*;
+
+    fn _create_oob() -> OutOfBandSender {
+        OutOfBandSender::create()
+            .set_label("oob-label")
+            .set_goal("issue-vc")
+            .set_goal_code(&GoalCode::IssueVC)
+    }
+
+    fn _create_service() -> ServiceResolvable {
+        ServiceResolvable::FullService(FullService::create()
+            .set_service_endpoint("http://example.org/agent".into())
+            .set_routing_keys(vec!("12345".into()))
+            .set_recipient_keys(vec!("abcde".into())))
+    }
+
+    #[test]
+    #[cfg(feature = "general_test")]
+    fn test_append_service_object_to_oob_services() {
+        let _setup = SetupMocks::init();
+
+        let service = _create_service();
+        let mut oob = _create_oob()
+            .append_service(&service);
+        let service_str = json!(service).to_string();
+        let resolved_service = oob.get_services();
+
+        assert_eq!(resolved_service.len(), 1);
+        assert_eq!(service, resolved_service[0]);
+    }
+
+    #[test]
+    #[cfg(feature = "general_test")]
+    fn test_append_did_to_oob_services() {
```

Comment:
I don't like this terminology - did is a type of service. Now that we have introduced this dychotomy, I would write `test_append_full_service_object_to_oob_services` and `test_append_did_service_object_to_oob_services`

Reply:
renamed